### ISSUE TITLE
Should resolveJsonModule resolve json files without a .json file extension?

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#24357]
**TypeScript Version:**  2.9.0-dev.20180521

**Search Terms:** 
- json
- resolveJsonModule

**Code**
In a project using `resolveJsonModule`, for the file:

```ts
import foo = require('./foo')
import foo2 = require('./foo.json')
 
console.log(foo, foo2);  
```

**Expected behavior:**
If we follow node, both imports are resolved

**Actual behavior:**
Only the `foo2` is resolved. The import without the `.json` file extension is not resolved

**Comment by CharlBest** [comment]
This would be very cool

**Comment by sheetalkamat** [comment]
I don't think we want to do this since we want developer to explicitly choose json file module resolution to avoid accidental includes and having no way to not include it.

**Comment by mnowaczyk** [comment]
This is causing me a headache. My app has to import a bunch of AVRO schemas kept in a common repo, which all have .avsc extension. Now thanks to your artificial restrictions ("to avoid accidental imports") I'm spending my time writing some silly workarounds instead of developing the actual app.

**Comment by fabioDMFerreira** [comment]
Any chance of having the json file resolve even if it doesn't have the extension?

I have an external dependency that didn't specify the extension. 

I think it should be a concern of the developer to pay attention to the collisions. Preferably, the typescript compiler should warn about the collisions, but I'm not sure about the performance/development implications of this.
